patchwork-fml/src/main/java/net/minecraftforge/fml/client/event/ConfigChangedEvent.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.client.event;
+
+import javax.annotation.Nullable;
+
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * These events are posted from the GuiConfig screen when the done button is pressed. The events are only posted
+ * if the parent screen is not an instance of GuiConfig or if the configID field has been set for
+ * the GuiConfig screen.
+ *
+ * <p>Listeners for this event should use OnConfigChanged or PostConfigChanged and check for a specific mod ID.
+ * For best results the listener should refresh any objects/fields that are set based on the mod's config
+ * and should serialize the modified config.</p>
+ *
+ * <p>TODO: These events will never be fired by Patchwork (until a ModMenu integration?)</p>
+ * @author bspkrs
+ */
+public class ConfigChangedEvent extends Event {
+	private final String modID;
+	private final boolean isWorldRunning;
+	private final boolean requiresMcRestart;
+	@Nullable
+	private final String configID;
+
+	public ConfigChangedEvent(String modID, @Nullable String configID, boolean isWorldRunning, boolean requiresMcRestart) {
+		this.modID = modID;
+		this.configID = configID;
+		this.isWorldRunning = isWorldRunning;
+		this.requiresMcRestart = requiresMcRestart;
+	}
+
+	@Override
+	public boolean hasResult() {
+		return true;
+	}
+
+	/**
+	 * The Mod ID of the mod whose configuration just changed.
+	 */
+	public String getModID() {
+		return modID;
+	}
+
+	/**
+	 * Whether or not a world is currently running.
```

Comment:
I know this is from Forge, but it's a bad description. Maybe just say:
```suggestion
	 * Whether or not the player is currently in a world.
```

Reply:
I *think* the description is meant to imply that it is only fired when there's a server running on the physical client, it's too vague to be sure. I'll leave it and add a `TODO: undefined behavior` line

Reply:
how about

`<p>TODO: This behavior is not well defined by Forge's original javadoc.</p>`

Reply:
That's more or less what I ended up putting.